src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -258,6 +262,67 @@ public function downloadVirtualFileAction(int $virtualProductFileId): BinaryFile
         return $response;
     }
 
+    /**
+     * @AdminSecurity("is_granted(['read'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param string $languageCode
+     *
+     * @return JsonResponse
+     *
+     * @throws ProductConstraintException
+     */
+    public function getAjaxProductsAction(Request $request, string $languageCode): JsonResponse
```

Comment:
I'm not a fan of this name.
- it contains "ajax" but I don't think a controller name should be used as an indicator of the usage
- it does not indicate it performs a search

May I suggest `searchProductsAction` or `searchForProductsAction` or `searchProductsForAssociationAction` ?

Reply:
I don't like it either, I didn't pick this at first but then I used the same naming as the category action that performs the same thing: https://github.com/PrestaShop/PrestaShop/blob/b9d4ce752c66528a063f5db4f1615d5d52bb6887/src/PrestaShopBundle/Controller/Admin/CategoryController.php#L132

But I just realized that this controller is quite old as it's in the Admin namespace, so I guess I shouldn't use it as a reference for naming I'll update the naming of the controller and route

Reply:
> I guess I shouldn't use it as a reference for naming I'll update the naming of the controller and route

👍